src/main/java/uk/gov/hmcts/reform/hmc/api/config/UserAuthTokenGenerator.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.hmcts.reform.hmc.api.config;
+
+import static org.springframework.http.HttpHeaders.EMPTY;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpServerErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+import uk.gov.hmcts.reform.hmc.api.exceptions.PrlUpdateException;
+import uk.gov.hmcts.reform.hmc.api.model.response.IdamTokenResponse;
+
+@Service
+public class UserAuthTokenGenerator {
+
+    @Value("${idam.refDataUserAuth.clientId}")
+    private String clientId;
+
+    @Value("${idam.refDataUserAuth.clientSecret}")
+    private String clientSecret;
+
+    @Value("${idam.refDataUserAuth.grantType}")
+    private String grantType;
+
+    @Value("${idam.refDataUserAuth.username}")
+    private String username;
+
+    @Value("${idam.refDataUserAuth.password}")
+    private String password;
+
+    @Value("${idam.refDataUserAuth.scope}")
+    private String scope;
+
+    RestTemplate restTemplateIdam = new RestTemplate();
+
+    @Value("${idam.refDataUserAuth.userIdamBaseurl}")
+    private String idamBaseUrl;
+
+    private static final String TOKEN_ENDPOINT_PATH = "/o/token";
```

Comment:
reuse the idam api services which already there

Reply:
resolved